src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -0,0 +1,161 @@
+import datetime
+import json
+import typing
+import uuid
+
+import inmanta.config
+import inmanta.protocol.common
+import inmanta_lsm.const
+import inmanta_lsm.model
+import pytest
+import pytest_inmanta.plugin
+
+# Error message to display when the lsm module is not reachable
+INMANTA_LSM_MODULE_NOT_LOADED = (
+    "The inmanta lsm module is not loaded.\n"
+    "    - If you are using v1 modules: make sure this code is called in a context where the project "
+    "fixture has been executed.\n"
+    "    - If you are using v2 modules: make sure the inmanta-module-lsm is installed in your venv."
+)
+
+
+def reset() -> None:
+    try:
+        # Import lsm module in function scope for usage with v1 modules
+        import inmanta_plugins.lsm
+    except ImportError as e:
+        raise RuntimeError(INMANTA_LSM_MODULE_NOT_LOADED) from e
+
+    # Reset the global cache as it used to be, but keep the client
+    inmanta_plugins.lsm.global_cache._instances_per_binding = {}
+    inmanta_plugins.lsm.global_cache._current_state_cache = {}
+
+
+class LsmProject:
+    def __init__(
+        self,
+        environment: uuid.UUID,
+        project: pytest_inmanta.plugin.Project,
+        monkeypatch: pytest.MonkeyPatch,
+        partial_compile: bool,
+    ) -> None:
+        inmanta.config.Config.set("config", "environment", str(environment))
+        self.services: typing.Dict[str, inmanta_lsm.model.ServiceInstance] = {}
+        self.project = project
+        self.monkeypatch = monkeypatch
+        self.partial_compile = partial_compile
+
+        try:
+            # Import lsm module in function scope for usage with v1 modules
+            import inmanta_plugins.lsm
+        except ImportError as e:
+            raise RuntimeError(INMANTA_LSM_MODULE_NOT_LOADED) from e
+
+        monkeypatch.setattr(inmanta_plugins.lsm.global_cache, "reset", reset)
+
+        self.monkeypatch.setattr(
+            inmanta_plugins.lsm.global_cache.get_client(),
+            "lsm_services_list",
+            self.lsm_services_list,
+            raising=False,
+        )
+
+        self.monkeypatch.setattr(
+            inmanta_plugins.lsm.global_cache.get_client(),
+            "lsm_services_update_attributes",
+            self.lsm_services_update_attributes,
+            raising=False,
+        )
+
+    @property
+    def environment(self) -> str:
+        return str(inmanta.config.Config.get("config", "environment"))
+
+    def lsm_services_list(self, tid: uuid.UUID, service_entity: str) -> inmanta.protocol.common.Result:
+        """
+        This is a mock for the lsm api, this method is called during allocation to get
+        all the instances of a service.
+        """
+        assert str(tid) == self.environment, f"{tid} != {self.environment}"
+        return inmanta.protocol.common.Result(
+            code=200,
+            result={
+                "data": [json.loads(srv.json()) for srv in self.services.values() if srv.service_entity == service_entity],
```

Comment:
if we round-trip serialize and load, shouldn't we use the inmanta custom encoder/decoder? 

If not, why not use `.dict()`

Reply:
Dict doesn't work because it won't serialize date correctly, can you point me to the inmanta custom encoder/decoder?

Reply:
https://github.com/inmanta/inmanta-core/blob/master/src/inmanta/util.py#L423

Reply:
I now use the inmanta json encoder, but the serialization and deserialization is still required to get rid of the python objects